starwars_api/models.py

```diff
@@ -55,27 +63,64 @@ def __repr__(self):
 class BaseQuerySet(object):
 
     def __init__(self):
-        pass
+        self.next_page = None
+        self.previous_page = None
+        self.results = []
+        self.current_record = 0
+        self.current_page = 0
 
     def __iter__(self):
-        pass
+        return self
 
     def __next__(self):
         """
         Must handle requests to next pages in SWAPI when objects in the current
         page were all consumed.
         """
-        pass
-
+        while True:
+            if not self.results:
+                self.current_page += 1
+                self.fetch_data(self.current_page)
+                self.current_record = 0
+            try:
+                return_data = self.results[self.current_record]
+            except IndexError:
+                self.current_page += 1
+                try:
+                    self.fetch_data(self.current_page)
+                except SWAPIClientError:
+                    raise StopIteration
+                self.current_record = 0
+                return_data = self.results[self.current_record]
+            self.current_record += 1
+            if self.RESOURCE_NAME == 'people':
+                return People(return_data)
+            elif self.RESOURCE_NAME == 'films':
+                return Films(return_data)
+        
     next = __next__
+    
+    def fetch_data(self, page_number = 1):
+        if self.RESOURCE_NAME == 'people':
+            json_data = api_client.get_people(**{'page': page_number})
+        elif self.RESOURCE_NAME == 'films':
+            json_data = api_client.get_films(**{'page': page_number})
+            
+        self.count = json_data['count']
+        self.next_page = json_data['next']
+        self.previous_page = json_data['previous']
+        self.results = json_data['results']
 
     def count(self):
         """
         Returns the total count of objects of current model.
         If the counter is not persisted as a QuerySet instance attr,
         a new request is performed to the API in order to get it.
         """
-        pass
+        _total = 0
+        for i in self:
+            _total += 1
+        return _total
```

Comment:
no need to reimplement like we did for LinkedLists. you defined self.count in init, use it. ^_^